### PR TITLE
EOS-19137: Reset offset when purge is called

### PR DESCRIPTION
**Pull request by selvakumaar5496**
Offsets are not refreshed if purge is called. Hence the get_unread_count() interface was returning a wrong count if purge(delete) is called. 
This PR contains the code which resets the offset if purge is being called.

Signed-off-by: Selvakumar <selva.k.nambi@seagate.com>

### CODE REVIEW

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 1]
```diff
@@ -323,15 +323,35 @@ def delete(self, admin_id: str, message_type: str):
...
+                    --bootstrap-server " + self._servers + " --group " \
+                    + consumer_group + " --topic " + message_type + \
+                    " --reset-offsets --to-latest --execute"
+                cmd_proc = SimpleProcess(cmd)
+                res_op, res_err, res_rc = cmd_proc.run()
+                if res_rc != 0:
+                    raise MessageBusError(errno.ENODATA, "Unable to reset the \
+                        offsets. %s", res_err)
+                decoded_string = res_op.decode('utf-8')
+                if 'Error' in decoded_string:
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:349]: I think "Error" will be string when res_rc != 0

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 2]
```diff
@@ -323,15 +323,35 @@ def delete(self, admin_id: str, message_type: str):
...
         Gets the count of unread messages from the Kafka message server
 
         Parameters:
+        message_type    This is essentially equivalent to the
+                        queue/topic name. For e.g. "Alert"
         consumer_group  A String that represents Consumer Group ID.
         """
         table = []
 
+        if self.get_log_size(message_type) == 0:
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:337]: Add comment on why we are doing this

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Check the comments.
Not seeing any test cases / or change in test case. Please ensure to have test cases for both classes producer, consumer.

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:338]: Added

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 1 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:350]: I get errors when rc is 0 as well. Hence checking the output(decoding bytes to string)

File: py-utils/src/utils/message_bus/message_broker_collection.py [hunk 3]
```diff
@@ -323,15 +323,36 @@ def delete(self, admin_id: str, message_type: str):
...
         Gets the count of unread messages from the Kafka message server
 
         Parameters:
+        message_type    This is essentially equivalent to the
+                        queue/topic name. For e.g. "Alert"
         consumer_group  A String that represents Consumer Group ID.
         """
         table = []
 
+        # Update the offsets if purge is called
```
Comment (sachinpunadikar) [inline py-utils/src/utils/message_bus/message_broker_collection.py:337]: The comment should be .. "if purge was called"

File: py-utils/test/test_message_bus/test_unread_count.py [hunk 4]
```diff
@@ -18,37 +18,70 @@
...
     """ Test MessageBus related functionality """
 
-    message_bus = MessageBus()
+    _msg_type = "biggg"
+    _consumer_group = "connn"
+    _msg_count = 100
+    _receive_limit = 10
 
-    def test_send(self):
+    def test_delete_count(self):
```
Comment (sachinpunadikar) [inline py-utils/test/test_message_bus/test_unread_count.py:32]: rename  test_delete_messages

File: py-utils/test/test_message_bus/test_unread_count.py [hunk 5]
```diff
@@ -18,37 +18,70 @@
...
+        """ Test Delete """
+        producer = MessageProducer(producer_id="p1", \
+            message_type=TestMessage._msg_type, method="sync")
+        producer.delete()
+        unread_count = producer.get_unread_count \
+            (consumer_group=TestMessage._consumer_group)
+        self.assertEqual(unread_count, 0)
+        self.assertFalse(unread_count > 0)
+
+    def test_send_count(self):
```
Comment (sachinpunadikar) [inline py-utils/test/test_message_bus/test_unread_count.py:42]: rename test_producer_read_count

File: py-utils/test/test_message_bus/test_unread_count.py [hunk 6]
```diff
@@ -18,37 +18,70 @@
...
+        self.assertEqual(len(messages), TestMessage._msg_count)
         self.assertIsInstance(messages, list)
         producer.send(messages)
+        unread_count = producer.get_unread_count(
+            consumer_group=TestMessage._consumer_group)
+        self.assertEqual(unread_count, TestMessage._msg_count)
+        self.assertFalse(unread_count != 100)
 
-    def test_count(self):
+    def test_receive_count(self):
```
Comment (sachinpunadikar) [inline py-utils/test/test_message_bus/test_unread_count.py:57]: rename test_consumer_read_count

File: py-utils/test/test_message_bus/test_unread_count.py [hunk 7]
```diff
@@ -18,37 +18,70 @@
...
         """ Test Unread Message Count """
```
Comment (sachinpunadikar) [inline py-utils/test/test_message_bus/test_unread_count.py:58]: Test Unread Message Count From Consumer side

File: py-utils/test/test_message_bus/test_unread_count.py [hunk 8]
```diff
@@ -18,37 +18,70 @@
...
+        producer = MessageProducer(producer_id="p1", \
+            message_type=TestMessage._msg_type, method="sync")
+        producer.delete()
+        unread_count = producer.get_unread_count \
+            (consumer_group=TestMessage._consumer_group)
+        self.assertEqual(unread_count, 0)
+        self.assertFalse(unread_count > 0)
+
+    def test_send_count(self):
         """ Test Send Message """
```
Comment (sachinpunadikar) [inline py-utils/test/test_message_bus/test_unread_count.py:43]: Test Unread Message Count From Producer side

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Please work on the comments, minor changes needed.

File: py-utils/src/utils/message_bus/message_broker_collection.py (hunk 3 above)
Comment (selvakumaar5496) [reply py-utils/src/utils/message_bus/message_broker_collection.py:337]: Done

File: py-utils/test/test_message_bus/test_unread_count.py (hunk 4 above)
Comment (selvakumaar5496) [reply py-utils/test/test_message_bus/test_unread_count.py:32]: renamed

File: py-utils/test/test_message_bus/test_unread_count.py (hunk 5 above)
Comment (selvakumaar5496) [reply py-utils/test/test_message_bus/test_unread_count.py:42]: renamed

File: py-utils/test/test_message_bus/test_unread_count.py (hunk 6 above)
Comment (selvakumaar5496) [reply py-utils/test/test_message_bus/test_unread_count.py:57]: renamed

File: py-utils/test/test_message_bus/test_unread_count.py (hunk 7 above)
Comment (selvakumaar5496) [reply py-utils/test/test_message_bus/test_unread_count.py:58]: Done

File: py-utils/test/test_message_bus/test_unread_count.py (hunk 8 above)
Comment (selvakumaar5496) [reply py-utils/test/test_message_bus/test_unread_count.py:43]: Done

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**